test/e2e/util.go

```diff
@@ -316,6 +336,62 @@ func buildEchoIngress(name types.NamespacedName, ingClass string, annotations ma
 		build()
 }
 
+// albcBuilder is a helper to build ALBC CRs.
+type albcBuilder struct {
+	nsname       types.NamespacedName
+	tagPolicy    albo.SubnetTaggingPolicy
+	ingressClass string
+	addons       []albo.AWSAddon
+	credentials  *albo.SecretReference
+}
+
+func newALBCBuilder() *albcBuilder {
+	return &albcBuilder{
+		tagPolicy:    albo.AutoSubnetTaggingPolicy,
+		ingressClass: "alb",
+		addons:       []albo.AWSAddon{},
+	}
+}
+
+func (b *albcBuilder) withName(name types.NamespacedName) *albcBuilder {
+	b.nsname = name
+	return b
+}
+
+func (b *albcBuilder) withIngressClass(class string) *albcBuilder {
+	b.ingressClass = class
+	return b
+}
+
+func (b *albcBuilder) withAddons(addons []albo.AWSAddon) *albcBuilder {
```

Comment:
You could use a variadic parameter to make things slightly more convenient for callers:
```suggestion
func (b *albcBuilder) withAddons(addons ...albo.AWSAddon) *albcBuilder {
```
Then, for example, this:
```go
	alb := newALBCBuilder().withName(name).withAddons([]albo.AWSAddon{albo.AWSAddonWAFv2, foo, bar, baz}).build()
```
becomes this:
```go
	alb := newALBCBuilder().withName(name).withAddons(albo.AWSAddonWAFv2, foo, bar, baz).build()
```

Reply:
Done.